### PR TITLE
feat: add transport compression option

### PR DESCRIPTION
**Pull request by StellarisW**
#### What type of PR is this?

feature

#### What this PR does / why we need it (English/Chinese):

add transport compression option

### CODE REVIEW

File: provider/options.go [hunk 1]
```diff
@@ -171,6 +172,17 @@ func WithInsecure() Option {
 	})
 }
 
+// WithCompress enable transport compression
+// It is the responsibility of the caller to ensure that the
+// compressor set has been registered with google.golang.org/grpc/encoding.
+// gzip can be registered by calling
+// `import _ "google.golang.org/grpc/encoding/gzip"`.
```
Comment (rogerogers) [inline provider/options.go:179]: Annotation is inaccurate

File: provider/options.go [hunk 2]
```diff
@@ -171,6 +172,17 @@ func WithInsecure() Option {
...
+func WithCompress() Option {
```
Comment (rogerogers) [inline provider/options.go:180]: might be withCompression?

File: provider/options.go (hunk 1 above)
Comment (StellarisW) [reply provider/options.go:179]: how? can you make it more accuratable?

**Review by rogerogers — APPROVED** [review]
LGTM

**Comment by li-jin-gou** [comment]
LGTM

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**